ee/audit/interceptor_ee.go

```diff
@@ -102,6 +117,40 @@ func AuditRequestHttp(next http.Handler) http.Handler {
 	})
 }
 
+func AuditWebSockets(ctx context.Context, req *schema.Request) {
+	if atomic.LoadUint32(&auditEnabled) == 0 {
+		return
+	}
+
+	namespace := uint64(0)
+	var user string
+	if token := req.Header.Get("X-Dgraph-AccessToken"); token != "" {
+		user = getUser(token, false)
+		namespace, _ = x.ExtractNamespaceFromJwt(token)
+	} else if token := req.Header.Get("X-Dgraph-AuthToken"); token != "" {
+		user = getUser(token, true)
+	} else {
+		user = getUser("", false)
+	}
+
+	ip := ""
+	if peerInfo, ok := peer.FromContext(ctx); ok {
+		ip, _, _ = net.SplitHostPort(peerInfo.Addr.String())
+	}
+
+	auditor.Audit(&AuditEvent{
+		User:        user,
+		Namespace:   namespace,
+		ServerHost:  x.WorkerConfig.MyAddr,
+		ClientHost:  ip,
+		Endpoint:    "/graphql",
+		ReqType:     WebSocket,
+		Req:         req.Query,
+		Status:      http.StatusText(http.StatusOK),
```

Comment:
Seems like we are setting the response status to OK in the audit event even before processing the request. Is it okay?
For example, It may be possible that the request may not have a valid token, even then we are setting an audit event to `ok`.

Reply:
graphql endpoints will always return 200 OK. thats why i put that